pkg/istio-agent/xds-proxy.go

```diff
@@ -141,17 +153,42 @@ func (p *XdsProxy) StreamAggregatedResources(downstream discovery.AggregatedDisc
 			// todo close downstream?
 			return err
 		case req := <-requestsChan:
+			if req.TypeUrl == v3.ListenerType {
+				// for every LDS req, add a NDS req
+				requestsChan <- &discovery.DiscoveryRequest{
```

Comment:
I think this can deadlock. If someone else sends on requestChan, it will be filled, and we will hang here forever.